### PR TITLE
mds: use variable g_ceph_context directly in MDSAuthCaps

### PR DESCRIPTION
**Pull request by rishabh-d-dave**
Variable g_ceph_context is global, therefore use it directly instead of
passing it as a parameter to method.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
  - [x] Code improvement
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by dparmar18** [comment]
the part of code looks much cleaner with this change

**Review by vshankar — APPROVED** [review]

**Comment by vshankar** [comment]
@rishabh-d-dave please pick this up in your next run.

**Comment by rishabh-d-dave** [comment]
@vshankar 
> @rishabh-d-dave please pick this up in your next run.

Yes.

**Comment by rishabh-d-dave** [comment]
No changes, just a rebase.

File: src/mon/AuthMonitor.cc [hunk 1]
```diff
@@ -1355,7 +1355,7 @@ bool AuthMonitor::valid_caps(
     }
   } else if (type == "mds") {
     MDSAuthCaps mdscap;
-    if (!mdscap.parse(g_ceph_context, caps, out)) {
+    if (!mdscap.parse(caps, out)) {
```
Comment (rzarzynski) [inline src/mon/AuthMonitor.cc:1358]: From the mon's perspective this should be fine.

Just a note:

> It is fine to use dout and derr in daemons, but in library code, you must use ldout and lderr, and pass in your own CephContext object. The compiler will enforce this restriction.
>
> Source: https://docs.ceph.com/en/latest/dev/context/

**Review by rzarzynski — APPROVED** [review]
The tiny mon change LGTM.
I hasn't checked the implicit assumption `MDSAuthCaps` isn't / won't be used in libraries.

**Comment by vshankar** [comment]
@rishabh-d-dave please rebase and do the needful.

**Comment by rishabh-d-dave** [comment]
> @rishabh-d-dave please rebase and do the needful.

Yes, I am on it. Due to teuthology infra issues, testing wasn't complete but it is now. Will review results once more and merge this PR.

**Pull request merged**